sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1676,6 +1676,23 @@ class Analyzer(
           // just return the original plan.
           case ae: AnalysisException => sort
         }
+      case sort @ Sort(sortOrders, _, child)
```

Comment:
I wonder if this case can be folded up into the previous case. They're not that different, right? The previous case handles direct child being `Aggregate` and this new case handles indirect child being `Aggregate`, but otherwise they should do similar things.

Reply:
Yes, we can probably collapse them, but it is non-trivial. This is mot likely related to your comment above. I'll try and do that, despite I am not entirely sure about the feasibility, I'll update you if I find some problems.